### PR TITLE
Allow specifying worker launcher and cmdline flags in the test

### PR DESCRIPTION
**Pull request by yuyichao**
Using environment variables like other test options so that it can be specified when running inside makefile.

The usecases I can currently think of are,
1. Running workers under `rr`, which I've been doing with [a much more specific patch](https://github.com/JuliaLang/julia/commit/9597f7e487c3d61a34e3949394490a1bedc2cbfd)
   
   Should also work with gdb, etc with the right options specified.
2. Running workers with specific options like bounds check/inlining/precompile/coverage.
   
   Ref https://github.com/staticfloat/julia-buildbot/pull/56, although I'm not 100% sure if we need to keep a no-inline buildbot.

These are meant to be expert options for specific cases so the default value is not appended (for the exeflags).

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -10,6 +10,18 @@ else
...
 end
 
+if haskey(ENV, "JULIA_TEST_EXEFLAGS")
+    const test_exeflags = `$(split(ENV["JULIA_TEST_EXEFLAGS"]))`
+else
+    const test_exeflags = `--check-bounds=yes --startup-file=no --depwarn=error`
+end
+
+if haskey(ENV, "JULIA_TEST_EXENAME")
+    const test_exename = `$(split(ENV["JULIA_TEST_EXENAME"]))`
```
Comment (tkelman) [inline test/runtests.jl]: should this be cleverer about shell-splitting if there's a `"path with spaces" --that-has-flags` ?

Comment (yuyichao) [inline test/runtests.jl]: Sure... seems that I can use a private function for that. I doubt it's common though but why not....

**Comment by yuyichao** [comment]
Any other comments?

**Comment by tkelman** [comment]
kind of a proliferation of undocumented environment variables that we're growing here, but I can see this being potentially useful

**Pull request merged**